apps/sms/js/thread_list_ui.js

```diff
@@ -408,89 +502,92 @@ var ThreadListUI = {
   // please make sure url will also be revoked if new delete api remove threads
   // without calling removeThread in the future.
   delete: function thlui_delete() {
-    function performDeletion() {
-      /* jshint validthis: true */
+    var selection = this.selectionHandler;
 
-      var threadIdsToDelete = [],
-          messageIdsToDelete = [],
-          threadCountToDelete = 0,
-          selected = this.selectionHandler.selectedList;
+    var promise = new Promise(function(resolve) {
+      ThreadListUI.performDeletion(selection.selectedList);
+      resolve();
+    });
 
-      function exitEditMode() {
-        ThreadListUI.cancelEdit();
-        WaitingScreen.hide();
-      }
+    promise.then(function() {
+      ThreadListUI.onDeletionReadUnread({
+        l10nId: 'deleted',
+        l10nArgs: {
+          n: selection.selectedCount
+        }
+      });
+    });
+  },
```

Comment:
ok, I think I understand what you want to do:
* you want to run performDeletion
* once it's finished, you want to run onDeletioNReadUnread

but this doesn't work.

What you really need is that performDeletion returns a promise :)